contracts/citycoin-token.clar

```diff
@@ -1,95 +1,120 @@
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-;; set constant for contract owner, used for updating token-uri
-(define-constant CONTRACT-OWNER tx-sender)
+(define-constant CONTRACT_OWNER tx-sender)
 
-;; define initial token URI
-(define-data-var token-uri (optional (string-utf8 256)) (some u"https://cdn.citycoins.co/metadata/citycoin.json"))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; ERROR CODES
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(define-map trusted-callers
-    principal
-    bool
-)
+(define-constant ERR_UNAUTHORIZED u3)
 
-;; set token URI to new value, only accessible by CONTRACT-OWNER
-(define-public (set-token-uri (new-uri (optional (string-utf8 256))))
-    (begin
-        (asserts! (is-eq tx-sender CONTRACT-OWNER) (err ERR-UNAUTHORIZED))
-        (ok (var-set token-uri new-uri))
-    )
-)
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; SIP-010 DEFINITION
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-;; The fungible token that can be Stacked.
-(define-fungible-token citycoins)
+(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.sip-010-trait)
+;; testnet: (impl-trait 'STR8P3RD1EHA8AA37ERSSSZSWKS9T2GYQFGXNA4C.sip-010-trait-ft-standard.sip-010-trait)
 
-;; SIP-010 functions
-(define-public (transfer (amount uint) (from principal) (to principal) (memo (optional (buff 34))))
-    (begin
-        (asserts! (is-eq from tx-sender)
-            (err ERR-UNAUTHORIZED))
+(define-fungible-token citycoins)
 
-        (if (is-some memo)
-            (print memo)
-            none
-        )
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; SIP-010 FUNCTIONS
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-        (ft-transfer? citycoins amount from to)
+(define-public (transfer (amount uint) (from principal) (to principal) (memo (optional (buff 34))))
+  (begin
+    (asserts! (is-eq from tx-sender) (err ERR_UNAUTHORIZED))
+    (if (is-some memo)
+      (print memo)
+      none
     )
+    (ft-transfer? citycoins amount from to)
+  )
 )
 
 (define-read-only (get-name)
-    (ok "citycoins"))
+  (ok "citycoins")
+)
 
 (define-read-only (get-symbol)
-    (ok "CYCN"))
+  (ok "CYCN")
+)
 
 (define-read-only (get-decimals)
-    (ok u0))
+  (ok u0)
+)
 
 (define-read-only (get-balance (user principal))
-    (ok (ft-get-balance citycoins user)))
+  (ok (ft-get-balance citycoins user))
+)
 
 (define-read-only (get-total-supply)
-    (ok (ft-get-supply citycoins)))
+  (ok (ft-get-supply citycoins))
+)
 
 (define-read-only (get-token-uri)
-    (ok (var-get token-uri)))
+  (ok (var-get tokenUri))
+)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; UTILITIES
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-;;---------------------------------------
-;; Other functionality
+(define-data-var tokenUri (optional (string-utf8 256)) (some u"https://cdn.citycoins.co/metadata/citycoin.json"))
+(define-data-var trustedCaller principal .citycoin-core)
```

Comment:
Q: should this be dynamic if `citycoin-core` ever changes?

Reply:
This is a good question. There are no plans for `citycoin-core` to change but we may want to provide a backup way to do this through `citycoin-auth` as part of #119.